lua/multicursors/utils.lua

```diff
@@ -493,4 +493,75 @@ M.exit = function()
     vim.b.MultiCursorSubLayer = nil
 end
 
+-- Generates hints based on the configuration and input parameters.
+---@param config Config configuration.
+---@param heads Head[]
+---@param mode string indicating the mode.
+---@return string hints as a string.
+M.generate_hints = function(config, heads, mode)
+    if config.generate_hints[mode] == false then
+        return 'MultiCursor ' .. mode .. ' mode'
+    elseif type(config.generate_hints[mode]) == 'string' then
+        return config.generate_hints[mode]
+    end
+
+    local num_items = #heads
+    local num_items_to_sort = num_items
+    local last_five_items = {}
+
+    -- Keep the last 5 items in the same pos.
+    -- Esc -> i -> c -> a -> e
+    if mode == 'normal' then
+        num_items_to_sort = math.max(num_items - 5, 0)
+        for i = num_items_to_sort + 1, num_items do
+            table.insert(last_five_items, heads[i])
+        end
+    end
+
+    ---@type Head[]
+    local items_to_sort = {}
+    for i = 1, num_items_to_sort do
+        table.insert(items_to_sort, heads[i])
+    end
+
+    table.sort(items_to_sort, function(a, b)
+        -- Custom sorting logic here (alphabetical sorting with special characters at the end)
+        local is_special_a = not string.match(a[1], '[%a%d]')
+        local is_special_b = not string.match(b[1], '[%a%d]')
+
+        if is_special_a and not is_special_b then
+            return false
+        elseif not is_special_a and is_special_b then
+            return true
+        else
+            return a[1] < b[1]
+        end
+    end)
+
+    ---@type Head[]
+    local sorted = {}
+    for _, value in ipairs(items_to_sort) do
+        table.insert(sorted, value)
+    end
+
+    for _, value in ipairs(last_five_items) do
+        table.insert(sorted, value)
+    end
+
+    -- NOTE:: The above code is not required if you do not want a fixed sort
+    -- Need to remove sorted
+    local str = ' MultiCursor: '
+        .. string.upper(string.sub(mode, 1, 1))
+        .. string.sub(mode, 2)
+        .. '\n'
+
+    for _, value in pairs(sorted) do
```

Comment:
if you could put more hints on the same line, it would be better.
right now it's a column taking whole screen space.

Reply:
Hmm, perhaps 2 columns ?

Reply:
Can we get screen width and put as many as we can fit?
if not 4 or 5.

Reply:
Hmm, Sure i can give this a shot.